Buffer Overflow in lzpack

Present on master@d9232579a11b8d9dceefdee308d54bd91979c85e

When making examples/n00bdemo I get the following error:

```
lzpack data.xml
LZPack v0.61b - File Compression and Packing Utility
2016-2019 Meido-Tek Productions (Lameguy64)

Creating textures.qlp in QLP format...
   Packing data/petscum16c.tim as petscum... Done.
   Packing data/bungirl.tim as bungirl... Done.
   Packing data/clktower.tim as clocktex... Done.
   Packing data/riftbld1.tim as riftbldg1... Done.
   Packing data/riftbld2.tim as riftbldg2... Done.
   Packing data/hatkid.tim as hatkid... Done.
   Packing data/celmapi.tim as celmap... Done.
   Packing data/lamelotl16c.tim as lamelotl... Done.
   Packing data/n00blogo-pixel.tim as n00blogo... Done.
   Packing data/font.tim as font... Done.
Packed 10 file(s) totaling 66900 bytes.
Creating data.lzp in LZP format...
   Packing data/mtekdisk.smd as mtekdisk... Ok. (45.62%)
   Packing data/mtektext.smd as mtektext... Ok. (46.07%)
   Packing data/star.smd as starsprite... Ok. (50.45%)
   Packing data/psn00blogo.smd as psn00blogo... Ok. (40.94%)
   Packing data/logo.smd as n00blogo... Ok. (37.08%)
   Packing data/petscum.smd as lightworld... Ok. (32.43%)
   Packing data/bulb.smd as lightbulb... Ok. (46.02%)
   Packing data/bungirl.smd as bungirl... Ok. (63.61%)
   Packing data/star_mask.smd as starmask... Ok. (52.73%)
   Packing data/timerift.smd as timerift... Ok. (47.88%)
   Packing data/rbowshade.smd as rbowshade... Ok. (44.23%)
   Packing data/hatkid.smd as hatkid... Ok. (54.04%)
   Packing textures.qlp as textures... Ok. (18.12%)
*** buffer overflow detected ***: lzpack terminated
makefile:38: recipe for target 'resources' failed
make: *** [resources] Aborted (core dumped)
```

GCC isn't wanting to give me debugging symbols, not sure why, so I can't give you the full backtrace. Here's what I have:
```
Reading symbols from ../../tools/bin/lzpack...(no debugging symbols found)...done.
[New LWP 27712]
Core was generated by `lzpack data.xml'.
Program terminated with signal SIGABRT, Aborted.
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
51	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
#1  0x00007f8263613801 in __GI_abort () at abort.c:79
#2  0x00007f826365c897 in __libc_message (action=action@entry=(do_abort | do_backtrace), fmt=fmt@entry=0x7f8263789988 "*** %s ***: %s terminated\n") at ../sysdeps/posix/libc_fatal.c:181
#3  0x00007f8263707cff in __GI___fortify_fail_abort (need_backtrace=need_backtrace@entry=true, msg=msg@entry=0x7f8263789905 "buffer overflow detected") at fortify_fail.c:33
#4  0x00007f8263707d21 in __GI___fortify_fail (msg=msg@entry=0x7f8263789905 "buffer overflow detected") at fortify_fail.c:44
#5  0x00007f8263705a10 in __GI___chk_fail () at chk_fail.c:28
#6  0x00007f8263704e49 in __strncpy_chk (s1=<optimized out>, s2=<optimized out>, n=<optimized out>, s1len=<optimized out>) at strncpy_chk.c:26
#7  0x00005626e74118dd in ?? ()
#8  0x00005626e741246f in ?? ()
#9  0x00005626e7411358 in ?? ()
#10 0x00007f82635f4b97 in __libc_start_main (main=0x5626e7411200, argc=2, argv=0x7fff20bead78, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7fff20bead68) at ../csu/libc-start.c:310
#11 0x00005626e74114ba in ?? ()
(gdb) 
```

I can't be 100% sure, but I think the issue is caused by https://github.com/Lameguy64/PSn00bSDK/blob/d9232579a11b8d9dceefdee308d54bd91979c85e/tools/lzpack/main.cpp#L232 . The strncpy call gives a buffer length of 4, but LZP_HEAD.id is a char[3]. I am working on a PR now.

System Details:

```
qbradq@qbradq-AB350-Gaming:~/PSn00bSDK/examples/n00bdemo$ uname --all
Linux qbradq-AB350-Gaming 4.15.0-45-generic #48-Ubuntu SMP Tue Jan 29 16:28:13 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
qbradq@qbradq-AB350-Gaming:~/PSn00bSDK/examples/n00bdemo$ gcc --version
gcc (Ubuntu 7.3.0-27ubuntu1~18.04) 7.3.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

qbradq@qbradq-AB350-Gaming:~/PSn00bSDK/examples/n00bdemo$ mipsel-unknown-elf-gcc --version
mipsel-unknown-elf-gcc (GCC) 9.1.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

--- a/tools/lzpack/main.cpp
+++ b/tools/lzpack/main.cpp
@@ -229,7 +229,7 @@ int CreateLZPfile(const char* packFile, FileListClass* fileList) {
 
     LZP_HEAD head;
 
-    strncpy(head.id, "LZP", 4);
+    strncpy(head.id, "LZP", sizeof(head.id));
     head.numFiles = fileList->EntryCount();
 
 	fseek(packp, 0, SEEK_SET);

